### ISSUE TITLE
CC0052 Make field readonly does not take ref out into consideration.

### ISSUE THREAD

**Issue by jmarolf** [code-cracker/code-cracker#788]
In the code below, CC0052 will fire, suggesting that `enableOption` can be made readonly.  

```C#
public class C
{
    private string field = "";
    private static void Foo(ref string bar) => bar = "";
    public void Baz() => Foo(ref field);
}
```

doing so would introduce error:
 
Severity | Code |	Description | Project | File | Line	
------------ | ------------- | ------------- | ------------- | ------------- | -------------
Error | CS0192 | A readonly field cannot be passed ref or out (except in a constructor) | MyProject.UI.Common | C:\MyProject\MyProject.UI.Common\ViewModels\OptionsViewModel.cs | 13

The PR for this item should go to branch `v1.0.x`.

**Comment by giggio** [comment]
Bug confirmed, I updated the code to a simpler case.

**Comment by giggio** [comment]
Closed on #817
